File: GFG/Enemy/Enemy.cpp
class Solution
{
    public:
        int largestArea(int n,int m,int k,vector<vector<int>> &enemy)
        {
            if(k==0) return n*m;
            vector<bool> r(n+1,false), c(m+1, false);
            
            
            for(int i=0;i<enemy.size();i++){
                int dr = enemy[i][0], dc = enemy[i][1];
                r[dr] = true, c[dc] = true;
            }
            int ans = 0, tmp = 0, dx = 0, dy = 0, mx = 0, my = 0;
            
            for(int i=1;i<=n;i++){
                if(r[i]==false) dx++;
                else dx = 0;
                mx = max(mx, dx);
            }
            
            for(int i=1;i<=m;i++){
                if(c[i]==false) dy++;
                else dy = 0;
                my = max(my, dy);
            }
            
            return mx*my;
        }
};

File: GFG/Enemy/Enemy.java
class Solution {
    public static int largestArea(int n,int m,int k,int[][] enemy) {
        // code here
        ArrayList<Integer> rows=new ArrayList<>();
        
        rows.add(-1);
        for(int[]i:enemy){
            rows.add(i[0]-1);
        }
        rows.add(n);
        Collections.sort(rows);
        
        int maxRow=0;
        for(int i=1;i<rows.size();i++){
            int prev=rows.get(i-1);
            int curr=rows.get(i);
            maxRow=Math.max(maxRow,curr-prev-1);
        }
        
        
        ArrayList<Integer> cols=new ArrayList<>();
        cols.add(-1);
        for(int[]i:enemy){
            cols.add(i[1]-1);
        }
        cols.add(m);
        Collections.sort(cols);
        
        
        int maxCol=0;
        for(int i=1;i<cols.size();i++){
            int prev=cols.get(i-1);
            int curr=cols.get(i);
            maxCol=Math.max(maxCol,curr-prev-1);
        }
        
        // System.out.println(maxRow);
        // System.out.println(maxCol);
        
        return maxRow*maxCol;
        
    }
}
 
